xcode/Subconscious/Shared/Models/FormField.swift

```diff
@@ -15,13 +15,31 @@ typealias FormFieldValidator<Input, Output> = (Input) -> Output?
 
 typealias FormFieldInput = Equatable & Hashable
 
-enum FormFieldAction<Input: FormFieldInput>: FormFieldInput {
+enum FormFieldAction<Input: FormFieldInput>:
+    FormFieldInput,
+    CustomStringConvertible
+{
     case reset
     /// Intended for triggering validation errors when a user submits a form containing this field
     case markAsTouched
     case focusChange(focused: Bool)
     case setValue(input: Input)
     case setValidationStatus(valid: Bool)
+
+    var description: String {
+        switch self {
+        case .reset:
+            return "reset"
+        case .markAsTouched:
+            return "markAsTouched"
+        case .focusChange(let focused):
+            return "focusChange(\(focused)"
+        case .setValue:
+            return "setValue(--redacted--)"
```

Comment:
Manually redact `setValue`. It's too hard to get away from strings for validated form fields, since most of our custom types return `nil` for invalid values, and we need the input to support intermediate values.

Validated form fields are more likely than other places to contain sensitive information which could leak through this action, so the easiest thing to do is redact all `setValue()` actions.

Reply:
Good call, I don't think we need to see the value anyway. It's always reflected on the UI in the context that these actions are generated.